### ISSUE TITLE
Reduce usage of afterEvaluate as much as possible

### ISSUE THREAD

**Issue by bmuschko** [wfhartford/gradle-helm#26]
**Problem:**

The plugin overuses `afterEvaluate` which causes issues if a user wants to make additional changes to task. At the moment this only possible if this is done in another `afterEvaluate` block in a build script.

**Solution:**

A lot of the `afterEvaluate` code be avoided by using the [lazy configuration concept](https://docs.gradle.org/current/userguide/lazy_configuration.html) aka `Provider` API.

**Comment by wfhartford** [comment]
I completely agree. I've used the providers and properties in more recent (non-public) gradle plugins and that is clearly a better solution. I've meant to give this plugin some love, but as I mentioned, it's a pretty low priority for me lately.
